### PR TITLE
Fix DSO resolution by reusing lookup done by bcc

### PR DESCRIPTION
**Pull request by marcin-ol**
Improve success rate for lookup of DSO names by reusing result of call to bcc `resolve_addr`.

### CODE REVIEW

File: examples/cpp/pyperf/PyPerfNativeStackTrace.cc [hunk 1]
```diff
@@ -94,20 +94,26 @@ NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
     unw_word_t offset;
     unw_word_t ip;
     struct bcc_symbol resolved_symbol;
+    resolved_symbol.module = nullptr;
     unw_get_reg(&cursor, UNW_REG_IP, &ip);
     if (procSymbols && procSymbols->resolve_addr(ip, &resolved_symbol, true)) {
         snprintf(buf, buf_size, "%s (%s)", resolved_symbol.demangle_name, resolved_symbol.module);
+        // let bcc take care of freeing memory allocated by __cxa_demangle()
+        bcc_symbol_free_demangle_name(&resolved_symbol);
```
Comment (Jongy) [inline examples/cpp/pyperf/PyPerfNativeStackTrace.cc:102]: Good call haha. 

I checked `get_stack_symbol` (which uses this API as well) and it seems to be the only "cleanup" function we should be using

**Comment by Jongy** [comment]
Not done reviewing

**Comment by marcin-ol** [comment]
Tested PyPerf with this change running for 30 minutes - found zero missing DSO names.

**Review by Jongy — APPROVED** [review]
Looks good. To make sure I understand what happens - BCC symbolication fails often, and in that case we automatically fell back to libunwind, but libunwind didn't include the DSO. So now when we fall to libunwind we still try to get the DSO from BCC?

I understand that now the success rate is higher, but still, why would BCC fail so often? As I described [here](https://github.com/Granulate/gprofiler/pull/579#issuecomment-1329925296)

**Comment by marcin-ol** [comment]
> Looks good. To make sure I understand what happens - BCC symbolication fails often, and in that case we automatically fell back to libunwind, but libunwind didn't include the DSO. So now when we fall to libunwind we still try to get the DSO from BCC?

That's correct. PyPerf falls back to libunwind.

> I understand that now the success rate is higher, but still, why would BCC fail so often? As I described [here](https://github.com/Granulate/gprofiler/pull/579#issuecomment-1329925296)
Bcc does use debug info files, if available. In general I suppose it's less thorough than libunwind in parsing the debug symbols. But can't yet explain what exactly does it skip, would need to research this in detail.
It could also be the address conversion issue bcc had, that was fixed in version 0.25.

**Pull request merged**